fairseq/modules/multihead_attention.py

```diff
@@ -150,6 +172,26 @@ def forward(
             and not torch.jit.is_scripting()
         ):
             assert key is not None and value is not None
+
+            if self.positional_embedding_layer is not None:
+                # even by recalculating queries, torch version is around
+                # 5% faster than fairseq version
+                q = self.q_proj(query)
+                q *= self.scaling
+                q = q.view(q.shape[0], -1, self.head_dim)
+                saved_state = None
+                if incremental_state is not None:
+                    saved_state = self._get_input_buffer(incremental_state)
+
+                positional_logits, _ = self.positional_embedding_layer(q, saved_state)
+
+                if attn_mask is not None:
+                    # The mask is added to "attention logits" before
+                    # softmax, and originally contains -inf to mask positions
+                    attn_mask = attn_mask + positional_logits
+                else:
+                    attn_mask = positional_logits
+
             return F.multi_head_attention_forward(
```

Comment:
Get shape mismatch error by adding these codes to fairseq, because torch's multi_head_attention_forward will expand the dim of attn_mask.

Reply:
hey,did u find the solution?